core/bin/snapshot_creator/src/main.rs

```diff
@@ -0,0 +1,115 @@
+use anyhow::Context as _;
+
+use zksync_dal::connection::DbVariant;
+use zksync_dal::ConnectionPool;
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_types::snapshots::{SnapshotChunk, SnapshotFactoryDependency, SnapshotStorageKey};
+use zksync_utils::ceil_div;
+
+async fn run(blob_store: Box<dyn ObjectStore>, pool: ConnectionPool) {
+    // TODO metrics
+    let mut conn = pool.access_storage().await.unwrap();
+
+    let l1_batch_number = conn
+        .blocks_dal()
+        .get_sealed_l1_batch_number()
+        .await
+        .unwrap();
+    let miniblock_number = conn
+        .storage_logs_snapshots_dal()
+        .get_last_miniblock_number(l1_batch_number)
+        .await
+        .unwrap();
+    let storage_logs_count = conn
+        .storage_logs_snapshots_dal()
+        .get_storage_logs_count(l1_batch_number)
+        .await
+        .unwrap();
+
+    //TODO load this from config
+    let chunk_size = 1_000_000;
+    let chunks_count = ceil_div(storage_logs_count, chunk_size);
+
+    tracing::info!(
+        "Creating snapshot for storage logs up to miniblock {}, l1_batch {}",
+        miniblock_number,
+        l1_batch_number.0
+    );
+    tracing::info!(
+        "{} chunks of max size {} will be generated",
+        chunks_count,
+        chunk_size
+    );
+
+    let mut output_files = vec![];
+    for chunk_id in 0..chunks_count {
+        let logs = conn
+            .storage_logs_snapshots_dal()
+            .get_storage_logs_chunk(l1_batch_number, chunk_id, chunk_size)
+            .await
+            .unwrap();
+        let mut factory_deps: Vec<SnapshotFactoryDependency> = vec![];
+        if chunk_id == 0 {
+            factory_deps = conn
+                .storage_logs_snapshots_dal()
+                .get_all_factory_deps(miniblock_number)
+                .await;
```

Comment:
do you suggest introducing separate StoredObject type for factory dependencies only? I wanted to avoid having separate "file type" for every type of object we need to store as it would introduce lots of additional complexity, but I agree it looks 'hacky' in the current form. It's also like that as I'm still not sure how big would the dependencies be on prod, it still may be required to introduce chunking of them as well.

Reply:
Got it.  @tomg10 thinking more about this - I think we decided to include data from initial_srites to storage_logs directly - that is, in the snapshot chunks. So you shouldnt need a separate file type, you'll just have more fields in each record (i think it's only enumeration_index).